### DISCUSSION TITLE
The link led (ledlink) is going off for half a second upon button press. Is there a way to prevent this?

### DISCUSSION THREAD

**Discussion by gitterman** [arendst/Tasmota#15728] [Q&A] [answered]
I have a couple of Sonoff and OBI power plugs which I have flashed with tasmota.
I have also build a custom device from scratch with an ESP Wroom chip.
On all devices the Wi-Fi link status LED goes off briefly when I press the button to toggle the relay.
It doesn't flash, though, if I toggle the relay from the web interface..
Is there an option to disable the flashing of the Wi-Fi LED when the button is pressed?

**Comment by sfromis** [comment]
[LedState](https://tasmota.github.io/docs/Commands/#ledstate)
[Status LEDs](https://tasmota.github.io/docs/Lights/#status-leds)

**Comment by gitterman** [comment]
Hmm,
from what the command description says, I do not understand the behaviour of the link status led nor how to configure it.
It says that LedState cannot be configured directly and mentions a relation to LedPower.
The link status LED is not the LED which shows the power state of the relay. 
There is a separate LED for that.
There is no hint that the LedLink flashes when the relay toggle button is pressed and not flashing, when the relay is toggled by the web interface. In my case the link status LED is ON steadily if the Wi-Fi link is established, but it goes OFF for about half a second if the relay is toggled by the associated button from either ON to OFF or vice versa.
It shines steadily, though, if the relay is toggled by the web GUI.

`LedPower<x> 	LED<x> power state control. Enabled only when LedLink(i) is configured
                0 = turn LED OFF and set LedState 0
                1 = turn LED ON and set LedState 0
                2 = toggle LED and set LedState 0
LedState        Manage LED state
                0 = disable use of LED as much as possible
                1 = show power state on LED (LED on when power on) (default) (inverted for Sonoff Touch/T1)
                2 = show MQTT subscriptions as a LED blink
                3 = show power state and MQTT subscriptions as a LED blink
                4 = show MQTT publications as a LED blink
                5 = show power state and MQTT publications as a LED blink
                6 = show all MQTT messages as a LED blink
                7 = show power state and MQTT messages as a LED blink
                8 = LED on when Wi-Fi and MQTT are connected.
Cannot be issued directly and is only activated when LedPower is switched from 0 to 1 due to a software function`

**Reply by barbudor** [reply]
You wrote it:
LedState 0 = disable use of LED as much as possible

**Comment by gitterman** [comment]
'LedState' initially gave  RESULT = {"LedState":1} and I already did change it to '0' with LedState 0, but the behaviour of the link status LED did not change at all.

**Comment by gitterman** [comment]
I have experimented a bit with the different settings of ledpower ledpower{x} and ledstate, but no success. 
I even checked the source files for ledlink, but did not find anything.
The link LED flashes when there is no link and glows steadily when a Wi-Fi link is established.
So far, so good.
But why does the link LED go off for about half a second when I press the button to toggle the relay?
This is a bug to me, as the Wi-Fi link is not affected by the button press.
As I said, the link LED stays on steadily when the relay is toggle from the web interface.

**Reply by ascillato** [reply]
Hi all these behaviors depend on the value of `SetOption31` and also how your GPIOs are configured.

If you want the led to inform only about Wi-Fi and MQTT connectivity, please, change that GPIO to be LEDLINK instead of LED1.

**Comment by gitterman** [comment]
I have a dedicated GPIO for the LedLinki function.
SetOption31 is 0 = Enabled (default)
If I disable it by setting it to 1, the LedLink doesn't flash anymore to indicate connection problems, which is bad.
BUT: The ledlink LED still goes off for half a second when I press the button.
That said, SetOption31 has no effect at all on the behavior of the link led with respect to the short drop out, when the relay button is pressed.

**Comment by arendst** [comment]
I simulated your situation and I see the ledlink flashing also.

I'll see when and why this sistuation came about.....

**Comment by gitterman** [comment]
Thanks for looking into it.
I found that the issue is already on my OBi devices with 6.5, so I suppose almost everybody has the effect.
I did not notice it myself until lately when I did build a switch box for my stereo with an esp wroom from scratch.
My box uses a more prominent LED for the ledlink function, and it is quite irritating to have the ledlink flash with every button press.

**Comment by arendst** [comment, marked as answer]
OK. I've gone back a year and also in there it acts as you found out.

Going over the code I conclude it's been there all the time! The goal was/is to show the user feedback of detected button press on the ledlink (as there are devices with more buttons and no leds under tasmota control).

So consider this as a feature.

**Comment by arendst** [comment]
If you do not want this blink I suggest you disable line 255 in file ``support_button.ino``.

**Reply by Jonsonmon** [reply]
it now called `support_button_v4.ino` ? the line 255 is:
`Button.last_state[i] = (digitalRead(Pin(GPIO_KEY1, i)) != bitRead(Button.inverted_mask, i));`
I Tried to disable this line and compile, but the blink is still there... can you please tell me which line should I Disable?

**Reply by sfromis** [reply]
Back then, line 255 was:
`          TasmotaGlobal.blinks = 201;`

**Reply by Jonsonmon** [reply]
Thanks

**Comment by gitterman** [comment]
Thanks for the solution.
I will give it a try.
Maybe it could become an option to disable this "feature" in future releases.

**Comment by Jonsonmon** [comment]
Please add an option to disable this annoying 'feature' .
